core-tests/shared/src/test/scala/zio/ZScheduleSpec.scala

```diff
@@ -234,7 +254,7 @@ object ZScheduleSpec
               ref <- Ref.make(0)
               i   <- alwaysFail(ref).retryOrElseEither(Schedule.once, ioFail)
             } yield i)
-              .foldM(
+              .foldM[Any, Int, String](
```

Comment:
Are all of these additional type annotations really necessary?

Reply:
I double checked and most of them are necessary because otherwise the inferred any will cause warnings